packages/dev/core/src/Materials/effectRenderer.ts

```diff
@@ -190,68 +191,193 @@ export class EffectRenderer {
+     */
+    blockCompilation?: boolean;
     /**
      * Callback when effect is compiled
      */
     onCompiled?: Nullable<(effect: Effect) => void>;
     /**
-     * The friendly name of the effect displayed in Spector.
+     * The friendly name of the effect (default: "effectWrapper")
      */
     name?: string;
     /**
      * The language the shader is written in (default: GLSL)
      */
     shaderLanguage?: ShaderLanguage;
+    /**
+     * Defines additional code to call to prepare the shader code
+     */
+    extraInitializations?: (useWebGPU: boolean, list: Promise<any>[]) => void;
     /**
      * Additional async code to run before preparing the effect
      */
     extraInitializationsAsync?: () => Promise<void>;
+    /**
+     * If the effect should be used as a post process (default: false). If true, the effect will be created with a "scale" uniform and a "textureSampler" sampler
+     */
+    useAsPostProcess?: boolean;
 }
 
 /**
  * Wraps an effect to be used for rendering
  */
 export class EffectWrapper {
     /**
-     * Event that is fired right before the effect is drawn (should be used to update uniforms)
+     * Force code to compile to glsl even on WebGPU engines.
+     * False by default. This is mostly meant for backward compatibility.
+     */
+    public static ForceGLSL = false;
+
+    private static _CustomShaderCodeProcessing: { [effectWrapperName: string]: EffectWrapperCustomShaderCodeProcessing } = {};
+
+    /**
+     * Registers a shader code processing with an effect wrapper name.
+     * @param effectWrapperName name of the effect wrapper. Use null for the fallback shader code processing. This is the shader code processing that will be used in case no specific shader code processing has been associated to an effect wrapper name
+     * @param customShaderCodeProcessing shader code processing to associate to the effect wrapper name
+     */
+    public static RegisterShaderCodeProcessing(effectWrapperName: Nullable<string>, customShaderCodeProcessing?: EffectWrapperCustomShaderCodeProcessing) {
+        if (!customShaderCodeProcessing) {
+            delete EffectWrapper._CustomShaderCodeProcessing[effectWrapperName ?? ""];
+            return;
+        }
+
+        EffectWrapper._CustomShaderCodeProcessing[effectWrapperName ?? ""] = customShaderCodeProcessing;
+    }
+
+    private static _GetShaderCodeProcessing(effectWrapperName: string) {
+        return EffectWrapper._CustomShaderCodeProcessing[effectWrapperName] ?? EffectWrapper._CustomShaderCodeProcessing[""];
+    }
+
+    /**
+     * Gets or sets the name of the effect wrapper
+     */
+    public get name() {
+        return this.options.name;
+    }
+
+    public set name(value: string) {
+        this.options.name = value;
+    }
+
+    /**
+     * Type of alpha mode to use when applying the effect (default: Engine.ALPHA_DISABLE)
+     */
+    public alphaMode = Constants.ALPHA_DISABLE;

```

Comment:
I am not sure alpha should be here ?

Reply:
If we want the `ThinXXXPostProcess` classes to have access to alpha mode, we must have this property in `EffectWrapper`, as the latter class is the base class for all `ThinXXXPostProcess` classes.